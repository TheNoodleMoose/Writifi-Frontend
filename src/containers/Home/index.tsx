import React from "react";
import styled from "styled-components";
import FeaturedCard, {
  FeaturedCardType
} from "../../components/Cards/FeaturedCards";
import Jumbotron from "../../components/Jumbotron";

const Home = () => {
  return (
    <HomeContainer>
      <Jumbotron />
      <Sidebar>
        <FeaturedCard type={FeaturedCardType.featured} />
        <FeaturedCard type={FeaturedCardType.editorsChoice} />
      </Sidebar>
    </HomeContainer>
  );
};

export default Home;

const HomeContainer = styled.div`
  width: 60%;
  margin: auto;
  display: flex;
  justify-content: space-around;

  @media (max-width: 768px) {
    width: 80%;
  }
`;

const Sidebar = styled.aside`
  width: 35%;
  @media (max-width: 768px) {
    display: none;
  }
`;
