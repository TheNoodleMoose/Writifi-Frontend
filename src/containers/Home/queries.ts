import gql from "graphql-tag";

export const GET_FEED = gql`
  query GetFeed {
    feed {
      id
      url
      description
    }
  }
`;
