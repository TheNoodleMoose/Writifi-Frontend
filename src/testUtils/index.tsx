import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
