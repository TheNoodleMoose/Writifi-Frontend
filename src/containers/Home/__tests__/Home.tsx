import React from "react";
import { renderWithContext } from "../../../testUtils";

import Home from "..";

it("renders correctly", () => {
  renderWithContext(<Home />).snapshot();
});
