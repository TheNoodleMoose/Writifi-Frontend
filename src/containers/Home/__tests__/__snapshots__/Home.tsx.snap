// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-fzqBZW llrcoi"
  >
    <div
      class="sc-fzoyAV jYYoNM"
    >
      <div
        class="sc-fzplWN gHhTD"
      >
        <div
          class="sc-fznyAO gqaNFL"
        >
          <svg
            height="200"
            width="200"
          >
            Logo.svg
          </svg>
          <div
            class="sc-fzokOt gcdrkZ"
          >
            <h1
              class="sc-fznZeY eqDpMx"
            >
              Endless stories at the tips of your fingers.
            </h1>
            <h1
              class="sc-fznZeY eqDpMx"
            >
              Create and Discover now.
            </h1>
          </div>
          <div
            class="sc-fznKkj jewRRC"
          >
            <a
              href="/signup"
            >
              <button
                class="sc-fzoLsD OgTwT"
                data-testid="Button"
              >
                <p
                  class="sc-fzpans jBkoMV"
                >
                  Start Discovering
                </p>
              </button>
            </a>
            <button
              class="sc-fzoLsD OgTwT"
              data-testid="Button"
            >
              <p
                class="sc-fzpans jBkoMV"
              >
                Why Writifi?
              </p>
            </button>
          </div>
        </div>
      </div>
      <aside
        class="sc-fzqNJr kcBmoJ"
      >
        <div
          class="sc-AxjAm jjKCsO"
        >
          <div
            class="sc-AxirZ fxlTtO"
            type="Featured"
          >
            <h2>
              Featured Reading
            </h2>
          </div>
          <div
            class="sc-AxiKw llAaWr"
          >
            <p
              class="sc-AxhCb cwwnQe"
              type="Featured"
            >
              Lorem ipsum dolor sit Molestias? Lorem ipsum dolor sit Molestias?
            </p>
            <p
              class="sc-AxhCb cwwnQe"
              type="Featured"
            >
              Lorem ipsum dolor sit Molestias?
            </p>
            <p
              class="sc-AxhCb cwwnQe"
              type="Featured"
            >
              Lorem ipsum dolor sit Molestias? Lorem ipsum dolor sit Molestias?Lorem ipsum dolor sit Molestias?
            </p>
            <p
              class="sc-AxhCb cwwnQe"
              type="Featured"
            >
              Lorem ipsum dolor sit Molestias?
            </p>
          </div>
        </div>
        <div
          class="sc-AxjAm jjKCsO"
        >
          <div
            class="sc-AxirZ kbkaQd"
            type="EditorsChoice"
          >
            <h2>
              Editor's Choice
            </h2>
          </div>
          <div
            class="sc-AxiKw llAaWr"
          >
            <p
              class="sc-AxhCb cwwnQe"
              type="EditorsChoice"
            >
              Lorem ipsum dolor sit Molestias? Lorem ipsum dolor sit Molestias?
            </p>
            <p
              class="sc-AxhCb cwwnQe"
              type="EditorsChoice"
            >
              Lorem ipsum dolor sit Molestias?
            </p>
            <p
              class="sc-AxhCb cwwnQe"
              type="EditorsChoice"
            >
              Lorem ipsum dolor sit Molestias? Lorem ipsum dolor sit Molestias?Lorem ipsum dolor sit Molestias?
            </p>
            <p
              class="sc-AxhCb cwwnQe"
              type="EditorsChoice"
            >
              Lorem ipsum dolor sit Molestias?
            </p>
          </div>
        </div>
      </aside>
    </div>
    <div
      class="sc-fzoLag hVCPnY"
    >
      <div
        class="sc-AxhUy fygIbw"
      >
        <h3
          class="sc-AxgMl cDHFlD"
        >
          My cool story about something...
        </h3>
        <p
          class="sc-AxheI bseEco"
        >
          This is the description of my story to somewhat sums it up...
        </p>
        <div
          class="sc-Axmtr iIIBwQ"
        >
          <h3
            class="sc-AxmLO lkchWc"
          >
            Christian Huffman
          </h3>
          <p
            class="sc-fzozJi fOLmBr"
          >
            @ some date in time
          </p>
        </div>
      </div>
      <div
        class="sc-AxhUy fygIbw"
      >
        <h3
          class="sc-AxgMl cDHFlD"
        >
          My cool story about something...
        </h3>
        <p
          class="sc-AxheI bseEco"
        >
          This is the description of my story to somewhat sums it up...
        </p>
        <div
          class="sc-Axmtr iIIBwQ"
        >
          <h3
            class="sc-AxmLO lkchWc"
          >
            Christian Huffman
          </h3>
          <p
            class="sc-fzozJi fOLmBr"
          >
            @ some date in time
          </p>
        </div>
      </div>
      <div
        class="sc-AxhUy fygIbw"
      >
        <h3
          class="sc-AxgMl cDHFlD"
        >
          My cool story about something...
        </h3>
        <p
          class="sc-AxheI bseEco"
        >
          This is the description of my story to somewhat sums it up...
        </p>
        <div
          class="sc-Axmtr iIIBwQ"
        >
          <h3
            class="sc-AxmLO lkchWc"
          >
            Christian Huffman
          </h3>
          <p
            class="sc-fzozJi fOLmBr"
          >
            @ some date in time
          </p>
        </div>
      </div>
      <div
        class="sc-AxhUy fygIbw"
      >
        <h3
          class="sc-AxgMl cDHFlD"
        >
          My cool story about something...
        </h3>
        <p
          class="sc-AxheI bseEco"
        >
          This is the description of my story to somewhat sums it up...
        </p>
        <div
          class="sc-Axmtr iIIBwQ"
        >
          <h3
            class="sc-AxmLO lkchWc"
          >
            Christian Huffman
          </h3>
          <p
            class="sc-fzozJi fOLmBr"
          >
            @ some date in time
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
